lib/_http_agent.js

```diff
@@ -168,6 +168,7 @@ Agent.prototype.addRequest = function addRequest(req, options, port/*legacy*/,
     var socket = this.freeSockets[name].shift();
     // Assign the handle a new asyncId and run any init() hooks.
     socket._handle.asyncReset();
+    socket[async_id_symbol] = socket._handle.getAsyncId();
```

Comment:
I don't know why but my intuition was to do this somewhere in a `Socket` method.
Maybe move both lines to a `Socket` method so other `Agent`s could use it.
Something like `Socket._prepareForReuse()`